frontend/ui/widget/filechooser.lua

```diff
@@ -221,7 +222,10 @@ function FileChooser:updateItems(select_number)
     self.path_items[self.path] = (self.page - 1) * self.perpage + (select_number or 1)
 end
 
-function FileChooser:refreshPath()
+function FileChooser:refreshPath(update_perpage)
+    if update_perpage then
+        self.perpage = G_reader_settings:readSetting("items_per_page")
+    end
```

Comment:
`G_reader_settings:readSetting()` costs nothing, so I guess you could do without any `update_perpage` argument.
And may be it will solve the fact that items_per_page is lost when you go from Classic mode (the item_per_page chosen) to Mosaic mode (hardcoded 3x3= 9 per_page) and back to classic_mode (you stay with 9 item per_page).

Reply:
Thanks, I also notice that (items_per_page is lost when you go from Classic mode (the item_per_page chosen) to Mosaic mode).
I didn't know that G_reader_settings:readSetting() costs nothing :)